src/utils/calcOdds.ts

```diff
@@ -138,14 +138,38 @@ type CalcPrematchOddsProps = {
   betAmount: string
   selections: Selection[]
   chainId: ChainId
+  isBatch?: boolean
 }
 
 export const calcPrematchOdds = async (props: CalcPrematchOddsProps): Promise<Record<string, number>> => {
-  const { config, selections, betAmount, chainId } = props
+  const { config, selections, betAmount, chainId, isBatch } = props
 
   const { betToken, contracts } = chainsData[chainId]
   const rawAmount = parseUnits(betAmount, betToken.decimals)
 
+  if (selections.length === 1 || isBatch) {
+    let odds: Record<string, number> = {}
+
+    for (let index = 0; index < selections.length; index++) {
+      const { conditionId, outcomeId, coreAddress } = selections[index]!
+
+      try {
+        const rawOdds = await readContract(config, {
```

Comment:
maybe readContracts to batch them into one request?

Reply:
672edf564525b2da10e4ae1b82579b3225725430

Reply:
bfbbc3b93bab6c91f46bef23437421406451c6e7